tests/unit/ProcessMaker/Nayra/FactoryTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use ProcessMaker\Nayra\Contracts\Factory;
+
+class FactoryTest extends TestCase
+{
+    /**
+     * Tests that the Factory instantiates classes whose constructors have and don't have arguments
+     */
+    public function testInstantiation ()
+    {
+        $config = $this->createMappingConfiguration();
+        $factory = new Factory($config);
+
+        $object1 = $factory->getInstanceOf(DummyInterface1::class);
+        $this->assertNotNull($object1);
```

Comment:
Add validation to check if the $object1 is an instance of the provided interface